% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
\documentclass[
  11pt,
]{article}
\usepackage{xcolor}
\usepackage[margin=1in]{geometry}
\usepackage{amsmath,amssymb}
\setcounter{secnumdepth}{5}
\usepackage{iftex}
\ifPDFTeX
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math} % this also loads fontspec
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
\usepackage{lmodern}
\ifPDFTeX\else
  % xetex/luatex font selection
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\newsavebox\pandoc@box
\newcommand*\pandocbounded[1]{% scales image to fit in text height/width
  \sbox\pandoc@box{#1}%
  \Gscale@div\@tempa{\textheight}{\dimexpr\ht\pandoc@box+\dp\pandoc@box\relax}%
  \Gscale@div\@tempb{\linewidth}{\wd\pandoc@box}%
  \ifdim\@tempb\p@<\@tempa\p@\let\@tempa\@tempb\fi% select the smaller of both
  \ifdim\@tempa\p@<\p@\scalebox{\@tempa}{\usebox\pandoc@box}%
  \else\usebox{\pandoc@box}%
  \fi%
}
% Set default figure placement to htbp
\def\fps@figure{htbp}
\makeatother
% definitions for citeproc citations
\NewDocumentCommand\citeproctext{}{}
\NewDocumentCommand\citeproc{mm}{%
  \begingroup\def\citeproctext{#2}\cite{#1}\endgroup}
\makeatletter
 % allow citations to break across lines
 \let\@cite@ofmt\@firstofone
 % avoid brackets around text for \cite:
 \def\@biblabel#1{}
 \def\@cite#1#2{{#1\if@tempswa , #2\fi}}
\makeatother
\newlength{\cslhangindent}
\setlength{\cslhangindent}{1.5em}
\newlength{\csllabelwidth}
\setlength{\csllabelwidth}{3em}
\newenvironment{CSLReferences}[2] % #1 hanging-indent, #2 entry-spacing
 {\begin{list}{}{%
  \setlength{\itemindent}{0pt}
  \setlength{\leftmargin}{0pt}
  \setlength{\parsep}{0pt}
  % turn on hanging indent if param 1 is 1
  \ifodd #1
   \setlength{\leftmargin}{\cslhangindent}
   \setlength{\itemindent}{-1\cslhangindent}
  \fi
  % set entry spacing
  \setlength{\itemsep}{#2\baselineskip}}}
 {\end{list}}
\usepackage{calc}
\newcommand{\CSLBlock}[1]{\hfill\break\parbox[t]{\linewidth}{\strut\ignorespaces#1\strut}}
\newcommand{\CSLLeftMargin}[1]{\parbox[t]{\csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLRightInline}[1]{\parbox[t]{\linewidth - \csllabelwidth}{\strut#1\strut}}
\newcommand{\CSLIndent}[1]{\hspace{\cslhangindent}#1}
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\usepackage{float}
\usepackage{booktabs}
\usepackage{longtable}
\usepackage{array}
\usepackage{multirow}
\usepackage{wrapfig}
\usepackage{colortbl}
\usepackage{pdflscape}
\usepackage{tabu}
\usepackage{threeparttable}
\usepackage{threeparttablex}
\usepackage[normalem]{ulem}
\usepackage{makecell}
\usepackage{xcolor}
\usepackage{bookmark}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\urlstyle{same}
\hypersetup{
  pdftitle={Task 3: RNA-seq Differential Gene Expression \& RNA-seq Functional Profiling},
  pdfauthor={Simon Safron {[}Mn: 11923407{]}, Alexander Veith {[}Mn: 12122739{]}, Miriam Überbacher {[}Mn:01627576{]}},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}

\title{Task 3: RNA-seq Differential Gene Expression \& RNA-seq
Functional Profiling}
\author{Simon Safron {[}Mn: 11923407{]}, Alexander Veith {[}Mn:
12122739{]}, Miriam Überbacher {[}Mn:01627576{]}}
\date{2025-12-11}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{2}
\tableofcontents
}
\newpage

\section{Introduction}\label{introduction}

\section{Raw Reads and Mapping QC}\label{raw-reads-and-mapping-qc}

\subsection{FastQC:}\label{fastqc}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\tightlist
\item
  According to FastQC: What was the minimum and the maximum number of
  read pairs sequenced per sample?
\end{enumerate}

\begin{figure}[H]

{\centering \includegraphics[width=0.85\linewidth]{fastqc_sequence_counts_plot} 

}

\caption{ Barplot of the number of read pairs per sample.}\label{fig:unnamed-chunk-1}
\end{figure}

\textbf{Answer:} As can be seen in Figure 1 the minimum number of reads
per sample is around 6.8 million and the maximum number of reads per
sample is around 15.7 million.

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{1}
\tightlist
\item
  What is the most overrepresented sequence (string of nucleotides) that
  was found by FastQC?
\end{enumerate}

\textbf{Answer:} According to the MultiQC report the most
overrepresented sequence was:

\begin{verbatim}
  "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA"
\end{verbatim}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{2}
\tightlist
\item
  What might be the reason there is so much of this specific
  sequence/homopolymer ?
\end{enumerate}

\textbf{Answer:} This could be from A tailed adapter dimers and PCR
slippage products that outcompete genuine RNA fragments in the library.

\subsection{HTSeq-count:}\label{htseq-count}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{3}
\tightlist
\item
  According to HTSeq Count: What was the minimum and the maximum number
  of read pairs reported per sample?
\end{enumerate}

\begin{figure}[H]

{\centering \includegraphics[width=0.85\linewidth]{htseq_assignment_plot} 

}

\caption{ Barplot of the number of read pairs, per sample HTSeq.}\label{fig:unnamed-chunk-2}
\end{figure}

\textbf{Answer:} As can be seen in Figure 2 the minimum number of reads
per sample is around 5.9 million and the maximum number of reads per
sample is around 14.8 million.

\newpage

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{5}
\tightlist
\item
  Which was the minimum and maximum percentage of reads uniquely
  assigned to a gene, as reported by HTSeq-count?
\end{enumerate}

\begin{figure}[H]

{\centering \includegraphics[width=0.85\linewidth]{htseq_assignment_plot_percentages} 

}

\caption{HTSeq assignment plot in percentages.}\label{fig:unnamed-chunk-3}
\end{figure}

\textbf{Answer:} As can be seen in Figure 3 the minimum percentage of
reads uniquely assigned to a gene is 61.3\% and the maximum percentage
of reads uniquely assigned to a gene is around 83.3\%.

\section{Dataset}\label{dataset}

\subsection{Raw read counts}\label{raw-read-counts}

\subsection{Sample information}\label{sample-information}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\tightlist
\item
  Which columns define the used base strain and substrain (WT or TGT
  mutant), respectively? Can you spot an error in one of those columns?
\end{enumerate}

\textbf{Answer:} The genotype column shows if the wild type or tgt
mutant was used. The strain column shows wich \emph{E.coli} strain was
used. The error is in the strain column suggesting that for every
experiment a different strain was used. But comparing this with the
strains and method supsection of the methods section of the paper shows
that only ``Escherichia coli K-12 MG1655 was used as the WT
strain.''.{[}1{]}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{1}
\tightlist
\item
  Which column defines if nickel was added to the media?
\end{enumerate}

\textbf{Answer:} The treatment column defines if nickel was added to the
media.

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{2}
\tightlist
\item
  Find the following information on the SRA Study page:
\end{enumerate}

\begin{quote}
Which type of Illumina machine was used for sequencing?
\end{quote}

\begin{quote}
\textbf{Answer:} Illumina HiSeq 2500
\end{quote}

\begin{quote}
What was the library layout?
\end{quote}

\begin{quote}
\textbf{Answer:} PAIRED
\end{quote}

\begin{quote}
When was the data released?
\end{quote}

\begin{quote}
\textbf{Answer:} 2022-05-16
\end{quote}

\section{Preprocessing of the data}\label{preprocessing-of-the-data}

\subsection{Filtering of the data}\label{filtering-of-the-data}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\tightlist
\item
  For how many genes did we originally retrieve count data?
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dim}\NormalTok{(rawCounts)}
\CommentTok{\#[1] 4295   12}
\end{Highlighting}
\end{Shaded}

\textbf{Answer:} Originally count data were retrieved for 4295 genes

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{1}
\tightlist
\item
  How many will be left after applying the filter?
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{dim}\NormalTok{(rawCounts[}\FunctionTok{rowSums}\NormalTok{(rawCounts) }\SpecialCharTok{\textgreater{}} \DecValTok{10}\NormalTok{, ])}
\CommentTok{\#[2] 4221   12}
\end{Highlighting}
\end{Shaded}

\textbf{Answer:} After applying the filter 4221 genes will be left.

\section{DGE Analysis}\label{dge-analysis}

\subsection{Differential Expression
Analysis}\label{differential-expression-analysis}

\subsection{Extracting results}\label{extracting-results}

Interpreting the summary:

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\tightlist
\item
  How many genes are significantly up-regulated and how many are
  significantly down-regulated in the nickel treated WT strain as
  compared to the untreated WT strain, using the default cutoff for the
  adjusted p-value?
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Extract results with default alpha (0.1)}
\NormalTok{DESeq2Results\_WT\_nickel }\OtherTok{\textless{}{-}} \FunctionTok{results}\NormalTok{(DESeq2Data, }
                                   \AttributeTok{contrast =} \FunctionTok{c}\NormalTok{(}\StringTok{"group"}\NormalTok{,}\StringTok{"WT.Nickel"}\NormalTok{,}\StringTok{"WT.none"}\NormalTok{))}

\CommentTok{\# View summary}
\FunctionTok{summary}\NormalTok{(DESeq2Results\_WT\_nickel)}


\CommentTok{\# out of 4221 with nonzero total read count}
\CommentTok{\# adjusted p{-}value \textless{} 0.1}
\CommentTok{\# LFC \textgreater{} 0 (up)       : 1069, 25\%}
\CommentTok{\# LFC \textless{} 0 (down)     : 1063, 25\%}
\CommentTok{\# outliers [1]       : 6, 0.14\%}
\CommentTok{\# low counts [2]     : 0, 0\%}
\CommentTok{\# (mean count \textless{} 1)}
\CommentTok{\# [1] see \textquotesingle{}cooksCutoff\textquotesingle{} argument of ?results}
\CommentTok{\# [2] see \textquotesingle{}independentFiltering\textquotesingle{} argument of ?results}
\end{Highlighting}
\end{Shaded}

\textbf{Answer:} 1069 genes are significantly up-regulated and 1063
genes are significatnly down-regulated in the nickel treated WT strain
as compared to the untreated WT strain, using the default cutoff for the
adjusted p-value of 10\%.

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{1}
\tightlist
\item
  What is the standard cutoff used for the significance level (adjusted
  p-value), if we don't change it?
\end{enumerate}

\textbf{Answer:} The standard cutoff used for the significance level
(adjusted p-value), if we don't change it is 10\% (p-value \textless{}
0.1).

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{2}
\tightlist
\item
  How many significantly differentially expressed genes does that make
  in total?
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Extract TGT.Nickel vs TGT.none with alpha = 0.1}
\NormalTok{DESeq2Results\_TGT\_nickel }\OtherTok{\textless{}{-}} \FunctionTok{results}\NormalTok{(DESeq2Data, }
                                    \AttributeTok{contrast =} \FunctionTok{c}\NormalTok{(}\StringTok{"group"}\NormalTok{,}\StringTok{"TGT.Nickel"}\NormalTok{,}\StringTok{"TGT.none"}\NormalTok{),}
                                    \AttributeTok{alpha =} \FloatTok{0.1}\NormalTok{)}

\CommentTok{\# View summary}
\FunctionTok{summary}\NormalTok{(DESeq2Results\_TGT\_nickel)}


\CommentTok{\# out of 4221 with nonzero total read count}
\CommentTok{\# adjusted p{-}value \textless{} 0.1}
\CommentTok{\# LFC \textgreater{} 0 (up)       : 986, 23\%}
\CommentTok{\# LFC \textless{} 0 (down)     : 877, 21\%}
\CommentTok{\# outliers [1]       : 6, 0.14\%}
\CommentTok{\# low counts [2]     : 164, 3.9\%}
\CommentTok{\# (mean count \textless{} 5)}
\CommentTok{\# [1] see \textquotesingle{}cooksCutoff\textquotesingle{} argument of ?results}
\CommentTok{\# [2] see \textquotesingle{}independentFiltering\textquotesingle{} argument of ?results}
\end{Highlighting}
\end{Shaded}

\textbf{Answer:} If we add up the up und down regulated genes we get the
total amount of significantly differentially expressed genes. Looking at
the summary this would be 1863 genes.

\begin{quote}
Changing the alpha factor:
\end{quote}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{3}
\tightlist
\item
  For the comparison of the genotypes under standard contitions. How
  many significantly differentially expressed genes in total are
  reported for a significance level of 0.05? (Go to the RStudio Help and
  search for ``results'' function, to identify the attribute you have to
  change.)
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Extract genotype comparison (TGT vs WT, no nickel) with alpha = 0.05}
\NormalTok{DESeq2Results\_genotype }\OtherTok{\textless{}{-}} \FunctionTok{results}\NormalTok{(DESeq2Data, }
                                  \AttributeTok{contrast =} \FunctionTok{c}\NormalTok{(}\StringTok{"group"}\NormalTok{,}\StringTok{"TGT.none"}\NormalTok{,}\StringTok{"WT.none"}\NormalTok{),}
                                  \AttributeTok{alpha =} \FloatTok{0.05}\NormalTok{)}

\CommentTok{\# View summary (this shows the numbers you need)}
\FunctionTok{summary}\NormalTok{(DESeq2Results\_genotype)}


\CommentTok{\# out of 4221 with nonzero total read count}
\CommentTok{\# adjusted p{-}value \textless{} 0.05}
\CommentTok{\# LFC \textgreater{} 0 (up)       : 187, 4.4\%}
\CommentTok{\# LFC \textless{} 0 (down)     : 275, 6.5\%}
\CommentTok{\# outliers [1]       : 6, 0.14\%}
\CommentTok{\# low counts [2]     : 0, 0\%}
\CommentTok{\# (mean count \textless{} 1)}
\CommentTok{\# [1] see \textquotesingle{}cooksCutoff\textquotesingle{} argument of ?results}
\CommentTok{\# [2] see \textquotesingle{}independentFiltering\textquotesingle{} argument of ?results}
\end{Highlighting}
\end{Shaded}

\textbf{Answer:} If we add up the up und down regulated genes we get the
total amount of significantly differentially expressed genes. Looking at
the summary this would be 462 genes at a p-value \textless{} 0.05.

Comparing the nickel treatment to no treatment in the TGT-mutant:

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{4}
\tightlist
\item
  Repeat the steps above for the comparison of the TGT-mutant strain
  treated with nickel to the TGT-mutant strain not treated with nickel.
  How many significantly differentially expressed genes in total are
  reported for a significance level of 0.05?
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Extract TGT mutant nickel effect with alpha = 0.05}
\NormalTok{DESeq2Results\_TGT\_nickel }\OtherTok{\textless{}{-}} \FunctionTok{results}\NormalTok{(DESeq2Data, }
                                    \AttributeTok{contrast =} \FunctionTok{c}\NormalTok{(}\StringTok{"group"}\NormalTok{,}\StringTok{"TGT.Nickel"}\NormalTok{,}\StringTok{"TGT.none"}\NormalTok{),}
                                    \AttributeTok{alpha =} \FloatTok{0.05}\NormalTok{)}

\CommentTok{\# View summary (this shows the numbers you need)}
\FunctionTok{summary}\NormalTok{(DESeq2Results\_TGT\_nickel)}

\CommentTok{\# out of 4221 with nonzero total read count}
\CommentTok{\# adjusted p{-}value \textless{} 0.05}
\CommentTok{\# LFC \textgreater{} 0 (up)       : 826, 20\%}
\CommentTok{\# LFC \textless{} 0 (down)     : 752, 18\%}
\CommentTok{\# outliers [1]       : 6, 0.14\%}
\CommentTok{\# low counts [2]     : 82, 1.9\%}
\CommentTok{\# (mean count \textless{} 3)}
\CommentTok{\# [1] see \textquotesingle{}cooksCutoff\textquotesingle{} argument of ?results}
\CommentTok{\# [2] see \textquotesingle{}independentFiltering\textquotesingle{} argument of ?results}
\end{Highlighting}
\end{Shaded}

\textbf{Answer:} If we add up the up und down regulated genes we get the
total amount of significantly differentially expressed genes. Looking at
the summary this would be 1578 genes at a p-value \textless{} 0.05.

\section{Vizualising data}\label{vizualising-data}

\subsection{Experimental QC - Clustering of samples
(PCA)}\label{experimental-qc---clustering-of-samples-pca}

\begin{figure}[H]

{\centering \includegraphics[width=0.85\linewidth]{PCAplot} 

}

\caption{Clustering of samples PCA dot plot.}\label{fig:unnamed-chunk-10}
\end{figure}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\tightlist
\item
  Do the groups of replicates behave as expected?
\end{enumerate}

\textbf{Answer:} Looking at the plot in figure 4 we can observe that the
individual groups are clearly separated from each other. It also makes
sense that there is a big distance between untreated and Nickel treated
strains (WT as well as TGT strains).

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{1}
\tightlist
\item
  Which sample would you identify as an outlier?
\end{enumerate}

\textbf{Answer:} Looking at figure 4 we would identify the WT strain
untreated as an outlier as it does not completely cluster with its
biological replicates but one point also clusters with the TGT untreated
cluster.

\subsection{Viewing counts for a single
geneID}\label{viewing-counts-for-a-single-geneid}

\begin{figure}[H]

{\centering \includegraphics[width=0.85\linewidth]{TGTcounts} 

}

\caption{Viewing counts for a single geneID in a dot plot.}\label{fig:unnamed-chunk-11}
\end{figure}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{2}
\tightlist
\item
  Think about the mutations in the E. coli strains and how that
  influences the transcripts of a gene. Are the read counts for the tgt
  gene in the wild type and the knockout strain what you expected?
  Explain why. \newpage
\end{enumerate}

\textbf{Answer:}

Wild type + none: tgt encodes tRNA guanine transglycosylase, an
essential enzyme for queuosine synthesis. It is expressed under normal
growth conditions to modify tRNAs. Expected Counts are going to be high.

Wild type + Nickel: tgt is transcriptionally repressed by nickel stress.
Expected Counts are going to be lower than that of Wt + none but still
higher than the expected counts of the knock out mutants.

tgt + none: The tgt gene is deleted so No functional tgt should be
transcribed. Expected counts are going to be very low to near zero.

tgt + Nickel: The tgt knock out stais the same but now Nickel is added.
This should not have a significant affect on the already very low to
near zero counts.

\section{Part 2: Functional Analysis and
Vizualisation}\label{part-2-functional-analysis-and-vizualisation}

\subsection{Setup}\label{setup}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{BiocManager}\SpecialCharTok{::}\FunctionTok{install}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"clusterProfiler"}\NormalTok{))}
\FunctionTok{install.packages}\NormalTok{(}\StringTok{"tidyverse"}\NormalTok{)}

\FunctionTok{install.packages}\NormalTok{(}\StringTok{"devtools"}\NormalTok{)}
\NormalTok{devtools}\SpecialCharTok{::}\FunctionTok{install\_github}\NormalTok{(}\StringTok{\textquotesingle{}kevinblighe/EnhancedVolcano\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(EnhancedVolcano)}
\FunctionTok{library}\NormalTok{(clusterProfiler)}
\FunctionTok{library}\NormalTok{(tidyverse)}
\FunctionTok{library}\NormalTok{(ggplot2)}
\FunctionTok{library}\NormalTok{(dplyr)}
\FunctionTok{library}\NormalTok{(DESeq2)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\DocumentationTok{\#\#\# load the data with read delim, because it is tab seperated}
\NormalTok{annotatedRawCounts }\OtherTok{\textless{}{-}} \FunctionTok{read\_delim}\NormalTok{(}\StringTok{"Counts\_raw.tsv"}\NormalTok{)}

\CommentTok{\#head(annotatedRawCounts)}

\NormalTok{annotatedRawCounts }\OtherTok{\textless{}{-}}\NormalTok{ annotatedRawCounts  }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{column\_to\_rownames}\NormalTok{(}\AttributeTok{var =} \StringTok{"ID"}\NormalTok{)}

\DocumentationTok{\#\#\# split data into rawCounts and Annotations}
\NormalTok{rawCounts }\OtherTok{\textless{}{-}}\NormalTok{ annotatedRawCounts[,}\DecValTok{3}\SpecialCharTok{:}\DecValTok{14}\NormalTok{]}
\NormalTok{annotations }\OtherTok{\textless{}{-}}\NormalTok{ annotatedRawCounts[,}\DecValTok{1}\SpecialCharTok{:}\DecValTok{2}\NormalTok{]}

\NormalTok{DESeq2ResultsDF }\OtherTok{\textless{}{-}} \FunctionTok{read\_delim}\NormalTok{(}\StringTok{"DESeq2Result\_treatment.tsv"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\section{Gene annotation (databases)}\label{gene-annotation-databases}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\tightlist
\item
  If you look closely at the 21 OrgDB packes, there are 2 different
  packages for E. coli. Which one should we use?
\end{enumerate}

\begin{figure}[H]

{\centering \includegraphics[width=0.85\linewidth]{database} 

}

\caption{Website of the used database for E.coli K12.}\label{fig:unnamed-chunk-15}
\end{figure}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{BiocManager}\SpecialCharTok{::}\FunctionTok{install}\NormalTok{(}\StringTok{"org.EcK12.eg.db"}\NormalTok{)}

\FunctionTok{library}\NormalTok{(org.EcK12.eg.db)}
\end{Highlighting}
\end{Shaded}

\textbf{Answer:} Since we are working with E. coli K12 MG1655 it would
make sense to use the genome wide annotation package for E.coli K12
(org.EcK12.eg.db)

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{2}
\tightlist
\item
  List the different available keytypes/identifiers.
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#gene names/identifiers}
\NormalTok{genes }\OtherTok{\textless{}{-}}\NormalTok{ annotations}\SpecialCharTok{$}\NormalTok{gene}
\FunctionTok{head}\NormalTok{(genes,}\DecValTok{10}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  [1] "thrL" "thrA" "thrB" "thrC" "yaaX" "yaaA" "yaaJ" "talB" "mog"  "satP"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#View all available keytypes}
\FunctionTok{keytypes}\NormalTok{(org.EcK12.eg.db)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  [1] "ACCNUM"      "ALIAS"       "ENTREZID"    "ENZYME"      "EVIDENCE"   
##  [6] "EVIDENCEALL" "GENENAME"    "GO"          "GOALL"       "ONTOLOGY"   
## [11] "ONTOLOGYALL" "PATH"        "PMID"        "REFSEQ"      "SYMBOL"
\end{verbatim}

\textbf{Answer:} The different keytypes and identifiers are shown above.

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{3}
\tightlist
\item
  Which of the keytypes/identifiers include the gene names given in the
  annotation?
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{keys }\OtherTok{\textless{}{-}} \FunctionTok{keys}\NormalTok{(org.EcK12.eg.db, }\AttributeTok{keytype =} \StringTok{"SYMBOL"}\NormalTok{)}
\FunctionTok{head}\NormalTok{(keys)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "yjhR"  "nfrA"  "thrL"  "insB1" "sspA"  "yaaJ"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#make sure our IDs are contained in the keys}
\CommentTok{\#prints the number of genes that are found within the keys}
\FunctionTok{sum}\NormalTok{(genes }\SpecialCharTok{\%in\%}\NormalTok{ keys)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 4274
\end{verbatim}

\textbf{Answer:} After iterating through all different keytypes, SYMBOL
was found to contain the genes.

\section{Running functional analysis}\label{running-functional-analysis}

\subsection{Overrepresentation
analysis}\label{overrepresentation-analysis}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\tightlist
\item
  How many significantly up- and downregulated genes are left, after
  applying the LFC cutoff?
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# filter our results by padj and log2FoldChange}
\NormalTok{res\_up }\OtherTok{\textless{}{-}}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{filter}\NormalTok{(DESeq2ResultsDF, padj }\SpecialCharTok{\textless{}} \FloatTok{0.05} \SpecialCharTok{\&}\NormalTok{ log2FoldChange }\SpecialCharTok{\textgreater{}} \DecValTok{1}\NormalTok{)}
\NormalTok{res\_down }\OtherTok{\textless{}{-}}\NormalTok{ dplyr}\SpecialCharTok{::}\FunctionTok{filter}\NormalTok{(DESeq2ResultsDF, padj }\SpecialCharTok{\textless{}} \FloatTok{0.05} \SpecialCharTok{\&}\NormalTok{ log2FoldChange }\SpecialCharTok{\textless{}} \SpecialCharTok{{-}}\DecValTok{1}\NormalTok{)}

\CommentTok{\# extract gene IDs for upregulated genes}
\NormalTok{genes\_up\_id }\OtherTok{\textless{}{-}}\NormalTok{ res\_up}\SpecialCharTok{$}\NormalTok{ID}
\NormalTok{genes\_up }\OtherTok{\textless{}{-}}\NormalTok{ annotations[genes\_up\_id,}\StringTok{"gene"}\NormalTok{]}

\CommentTok{\# extract gene names for downregulated genes}
\NormalTok{genes\_down\_id }\OtherTok{\textless{}{-}}\NormalTok{ res\_down}\SpecialCharTok{$}\NormalTok{ID}
\NormalTok{genes\_down }\OtherTok{\textless{}{-}}\NormalTok{ annotations[genes\_down\_id,}\StringTok{"gene"}\NormalTok{]}

\CommentTok{\# append both lists to get all deregulated genes}
\NormalTok{genes\_de }\OtherTok{\textless{}{-}} \FunctionTok{c}\NormalTok{(genes\_up,genes\_down)}

\FunctionTok{length}\NormalTok{(genes\_up)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 660
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(genes\_down)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 582
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{length}\NormalTok{(genes\_de)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 1242
\end{verbatim}

\textbf{Answer:} With the \emph{length} command we can show the number
of genes up- or downregulated. In total, 660 genes were up- and 582
genes were downregulated. In sum, 1242 genes are deregulated.

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{1}
\tightlist
\item
  How many significantly over-represented biological processes (GO
  terms) are there, per subset of genes (all differentially expressed
  genes, up-regulated genes only, down-regulated genes only.)
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{EC }\OtherTok{\textless{}{-}} \StringTok{"org.EcK12.eg.db"}
\NormalTok{EC\_KEY }\OtherTok{\textless{}{-}} \StringTok{"SYMBOL"}

\NormalTok{orBP }\OtherTok{\textless{}{-}} \FunctionTok{enrichGO}\NormalTok{(genes\_de, }
\NormalTok{                 EC, }
                 \AttributeTok{ont=}\StringTok{"BP"}\NormalTok{,}
                 \AttributeTok{keyType =}\NormalTok{ EC\_KEY,}
                 \AttributeTok{pvalueCutoff=}\FloatTok{0.05}\NormalTok{) }


\NormalTok{orUpBP }\OtherTok{\textless{}{-}} \FunctionTok{enrichGO}\NormalTok{(genes\_up, }
\NormalTok{                 EC, }
                 \AttributeTok{ont=}\StringTok{"BP"}\NormalTok{, }
                 \AttributeTok{keyType =}\NormalTok{ EC\_KEY, }
                 \AttributeTok{pvalueCutoff=}\FloatTok{0.05}\NormalTok{) }


\NormalTok{orDownBP }\OtherTok{\textless{}{-}} \FunctionTok{enrichGO}\NormalTok{(genes\_down, }
\NormalTok{                 EC, }
                 \AttributeTok{ont=}\StringTok{"BP"}\NormalTok{,}
                 \AttributeTok{keyType =}\NormalTok{ EC\_KEY,}
                 \AttributeTok{pvalueCutoff=}\FloatTok{0.05}\NormalTok{)}

\CommentTok{\#To show the number of statistically significant GO terms}
\NormalTok{n\_BP\_all }\OtherTok{\textless{}{-}} \FunctionTok{nrow}\NormalTok{(orBP)}
\NormalTok{n\_BP\_up }\OtherTok{\textless{}{-}} \FunctionTok{nrow}\NormalTok{(orUpBP)}
\NormalTok{n\_BP\_down }\OtherTok{\textless{}{-}} \FunctionTok{nrow}\NormalTok{(orDownBP)}

\FunctionTok{print}\NormalTok{(n\_BP\_all)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 100
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(n\_BP\_up)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 59
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(n\_BP\_down)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 93
\end{verbatim}

\textbf{Answer:} There are in summary 100 GO terms deregulated, 59 up
and 93 down. For a quick look into the first few results, take a look at
the tables below.

\begin{table}[H]
\centering
\caption{\label{tab:unnamed-chunk-21}Top 5 overrepresented BP terms - All DE genes}
\centering
\resizebox{\ifdim\width>\linewidth\linewidth\else\width\fi}{!}{
\fontsize{9}{11}\selectfont
\begin{tabular}[t]{llrr}
\toprule
  & Description & p.adjust & Count\\
\midrule
GO:0006935 & chemotaxis & 0 & 34\\
GO:0042330 & taxis & 0 & 34\\
GO:0040011 & locomotion & 0 & 42\\
GO:0001539 & cilium or flagellum-dependent cell motility & 0 & 49\\
GO:0071973 & bacterial-type flagellum-dependent cell motility & 0 & 49\\
\bottomrule
\end{tabular}}
\end{table}

\begin{table}

\caption{\label{tab:unnamed-chunk-21}Overrepresented BP terms - Upregulated genes}
\centering
\begin{tabular}[t]{l|l|r|r}
\hline
  & Description & p.adjust & Count\\
\hline
GO:0046377 & colanic acid metabolic process & 0.00e+00 & 18\\
\hline
GO:0009242 & colanic acid biosynthetic process & 2.00e-07 & 15\\
\hline
GO:0072348 & sulfur compound transport & 4.00e-07 & 23\\
\hline
GO:0006857 & oligopeptide transport & 2.20e-06 & 21\\
\hline
GO:0042938 & dipeptide transport & 3.42e-05 & 15\\
\hline
\end{tabular}
\end{table}

\begin{table}

\caption{\label{tab:unnamed-chunk-21}Overrepresented BP terms - Downregulated genes}
\centering
\begin{tabular}[t]{l|l|r|r}
\hline
  & Description & p.adjust & Count\\
\hline
GO:0006935 & chemotaxis & 0 & 32\\
\hline
GO:0042330 & taxis & 0 & 32\\
\hline
GO:0001539 & cilium or flagellum-dependent cell motility & 0 & 42\\
\hline
GO:0071973 & bacterial-type flagellum-dependent cell motility & 0 & 42\\
\hline
GO:0097588 & archaeal or bacterial-type flagellum-dependent cell motility & 0 & 42\\
\hline
\end{tabular}
\end{table}

\section{Vizualization}\label{vizualization}

\subsection{Vulcano Plot}\label{vulcano-plot}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\tightlist
\item
  Which is the most down-regulated gene and which is the most
  up-regulated gene, in terms of LFC?
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Set a boolean column for significance}
\NormalTok{DESeq2ResultsDF}\SpecialCharTok{$}\NormalTok{significant }\OtherTok{\textless{}{-}} \FunctionTok{ifelse}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(DESeq2ResultsDF}\SpecialCharTok{$}\NormalTok{padj) }\SpecialCharTok{\&}\NormalTok{ DESeq2ResultsDF}\SpecialCharTok{$}\NormalTok{padj }\SpecialCharTok{\textless{}}\NormalTok{ .}\DecValTok{05}\NormalTok{ , T, F)}

\DocumentationTok{\#\# Add gene annotations}
\NormalTok{DESeq2ResultsDF}\SpecialCharTok{$}\NormalTok{gene }\OtherTok{\textless{}{-}}\NormalTok{ annotations[DESeq2ResultsDF}\SpecialCharTok{$}\NormalTok{ID,}\StringTok{"gene"}\NormalTok{]}

\FunctionTok{EnhancedVolcano}\NormalTok{(DESeq2ResultsDF,}
                \AttributeTok{lab =}\NormalTok{  DESeq2ResultsDF}\SpecialCharTok{$}\NormalTok{gene,}
                \AttributeTok{x =} \StringTok{\textquotesingle{}log2FoldChange\textquotesingle{}}\NormalTok{,}
                \AttributeTok{y =} \StringTok{\textquotesingle{}padj\textquotesingle{}}\NormalTok{,}
                \AttributeTok{pCutoff =} \FloatTok{0.05}\NormalTok{)}\CommentTok{\# + xlim(0,2.5) + ylim(0,10) you can further restrict the plot by setting limits on x an y axis}
\end{Highlighting}
\end{Shaded}

\begin{center}\includegraphics{Task-3_files/figure-latex/unnamed-chunk-22-1} \end{center}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#show the entry with the highest log2FoldChange}
\NormalTok{DESeq2ResultsDF[DESeq2ResultsDF}\SpecialCharTok{$}\NormalTok{log2FoldChange }\SpecialCharTok{==} \FunctionTok{max}\NormalTok{(DESeq2ResultsDF}\SpecialCharTok{$}\NormalTok{log2FoldChange),]}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 1 x 9
##   baseMean log2FoldChange lfcSE  stat   pvalue     padj ID    significant gene 
##      <dbl>          <dbl> <dbl> <dbl>    <dbl>    <dbl> <chr> <lgl>       <chr>
## 1     41.6           8.82  1.20  7.33 2.25e-13 2.33e-12 b4721 TRUE        ytiD
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#show the entry with the lowest log2FoldChange}
\NormalTok{DESeq2ResultsDF[DESeq2ResultsDF}\SpecialCharTok{$}\NormalTok{log2FoldChange }\SpecialCharTok{==} \FunctionTok{min}\NormalTok{(DESeq2ResultsDF}\SpecialCharTok{$}\NormalTok{log2FoldChange),]}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## # A tibble: 1 x 9
##   baseMean log2FoldChange lfcSE  stat   pvalue     padj ID    significant gene 
##      <dbl>          <dbl> <dbl> <dbl>    <dbl>    <dbl> <chr> <lgl>       <chr>
## 1   38216.          -7.78 0.378 -20.6 7.41e-94 3.13e-91 b4354 TRUE        btsT
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ggsave}\NormalTok{(}\StringTok{\textquotesingle{}figs/volcano\_plot.png\textquotesingle{}}\NormalTok{, }\AttributeTok{width =} \DecValTok{12}\NormalTok{, }\AttributeTok{height =} \DecValTok{8}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\textbf{Answer:} The most upregulated gene is \emph{ytiD} with a L2FC of
8.82. The most downregulated gene is \emph{btsT} with a L2FC of -7.78.

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{1}
\tightlist
\item
  How many times are those genes higher or lower expressed in treated
  compared to untreated WT strain? (Calculate the fold change from the
  LFC)
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{DESeq2ResultsDF }\OtherTok{\textless{}{-}} \FunctionTok{read\_delim}\NormalTok{(}\StringTok{"DESeq2Result\_treatment.tsv"}\NormalTok{)}
\NormalTok{res\_df }\OtherTok{\textless{}{-}} \FunctionTok{as.data.frame}\NormalTok{(DESeq2ResultsDF)}

\NormalTok{most\_up\_idx }\OtherTok{\textless{}{-}} \FunctionTok{which.max}\NormalTok{(res\_df}\SpecialCharTok{$}\NormalTok{log2FoldChange)}
\NormalTok{most\_up }\OtherTok{\textless{}{-}}\NormalTok{ res\_df[most\_up\_idx, ]}

\NormalTok{most\_down\_idx }\OtherTok{\textless{}{-}} \FunctionTok{which.min}\NormalTok{(res\_df}\SpecialCharTok{$}\NormalTok{log2FoldChange)}
\NormalTok{most\_down }\OtherTok{\textless{}{-}}\NormalTok{ res\_df[most\_down\_idx, ]}

\NormalTok{gene\_name\_up }\OtherTok{\textless{}{-}}\NormalTok{ annotations[most\_up}\SpecialCharTok{$}\NormalTok{ID, }\StringTok{"gene"}\NormalTok{]}
\NormalTok{gene\_name\_down }\OtherTok{\textless{}{-}}\NormalTok{ annotations[most\_down}\SpecialCharTok{$}\NormalTok{ID, }\StringTok{"gene"}\NormalTok{]}


\NormalTok{most\_up}\SpecialCharTok{$}\NormalTok{foldChange }\OtherTok{\textless{}{-}} \DecValTok{2}\SpecialCharTok{\^{}}\NormalTok{most\_up}\SpecialCharTok{$}\NormalTok{log2FoldChange}
\NormalTok{most\_down}\SpecialCharTok{$}\NormalTok{foldChange }\OtherTok{\textless{}{-}}  \DecValTok{2}\SpecialCharTok{\^{}}\NormalTok{most\_down}\SpecialCharTok{$}\NormalTok{log2FoldChange}

\NormalTok{most\_up}\SpecialCharTok{$}\NormalTok{gene }\OtherTok{\textless{}{-}}\NormalTok{ gene\_name\_up}
\NormalTok{most\_down}\SpecialCharTok{$}\NormalTok{gene }\OtherTok{\textless{}{-}}\NormalTok{ gene\_name\_down}

\FunctionTok{print}\NormalTok{(}\FunctionTok{c}\NormalTok{(most\_up}\SpecialCharTok{$}\NormalTok{log2foldChange, most\_up}\SpecialCharTok{$}\NormalTok{foldChange ,most\_up}\SpecialCharTok{$}\NormalTok{gene ))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "451.110505802606" "ytiD"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{print}\NormalTok{(}\FunctionTok{c}\NormalTok{(most\_down}\SpecialCharTok{$}\NormalTok{log2foldChange, most\_down}\SpecialCharTok{$}\NormalTok{foldChange, most\_down}\SpecialCharTok{$}\NormalTok{gene))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "0.00455492673103271" "btsT"
\end{verbatim}

\textbf{Answer:} For a comprehensive comparison, we can do this manually
by thinking of how the LFC is defined. Since it's a logarithmic function
with base 2, we can calculate the FoldChange: FC = 2\^{}LFC and compare
the results manually. For faster and automated calculation, one may use
the code above, which takes the data from the DESeq dataset for treated
and untreated WT and calculates the FC. Gene \emph{ytiD} is 451.1 times
higher transcribed than in untreated WT. Gene \emph{btsT} is 217.4 times
lower transcribed than in untreated WT.

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{2}
\tightlist
\item
  Which of the genes shows the most significant deregulation (the lowest
  p-value)?
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{min\_pvalue\_idx }\OtherTok{\textless{}{-}} \FunctionTok{which.min}\NormalTok{(res\_df}\SpecialCharTok{$}\NormalTok{pvalue)}
\NormalTok{gene\_min\_pvalue }\OtherTok{\textless{}{-}}\NormalTok{ res\_df[min\_pvalue\_idx, ]}

\NormalTok{gene\_id\_sign }\OtherTok{\textless{}{-}}\NormalTok{ gene\_min\_pvalue}\SpecialCharTok{$}\NormalTok{ID}
\NormalTok{gene\_name\_sign }\OtherTok{\textless{}{-}}\NormalTok{ annotations[gene\_id\_sign, }\StringTok{"gene"}\NormalTok{]}

\NormalTok{most\_sign }\OtherTok{\textless{}{-}}\NormalTok{ gene\_min\_pvalue}
\NormalTok{most\_sign}\SpecialCharTok{$}\NormalTok{gene }\OtherTok{\textless{}{-}}\NormalTok{ gene\_name\_sign}

\FunctionTok{print}\NormalTok{(}\FunctionTok{c}\NormalTok{(gene\_min\_pvalue}\SpecialCharTok{$}\NormalTok{pvalue, most\_sign}\SpecialCharTok{$}\NormalTok{gene))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] "5.66136528139081e-124" "fliM"
\end{verbatim}

\textbf{Answer:} The gene with the lowest p-value (the most significant
deregulation): \emph{fliM} with a \textbf{p-value = 5.66e-124}.

\subsection{Dotplot}\label{dotplot}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\tightlist
\item
  Which is the most significantly enriched GO term ?
\end{enumerate}

\begin{center}\includegraphics{Task-3_files/figure-latex/unnamed-chunk-25-1} \end{center}

\begin{center}\includegraphics{Task-3_files/figure-latex/unnamed-chunk-25-2} \end{center}

\begin{center}\includegraphics{Task-3_files/figure-latex/unnamed-chunk-25-3} \end{center}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\tightlist
\item
  Which is the most significantly enriched GO term in terms of
  upregulated genes?
\end{enumerate}

\textbf{Answer:} By looking at the graphs,which are sorted by adjusted
p-values, we can see that the \textbf{genes for response to starvation}
have the lowest p-value, therefore are the most significant upregulated
GO term.

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{1}
\tightlist
\item
  Which is the most significantly enriched GO term in terms of
  downregulated genes?
\end{enumerate}

\textbf{Answer:} By looking at the graphs,which are sorted by adjusted
p-values, we can see that the \textbf{genes for iron import into the
cell} have the lowest p-value, therefore are the most significant
downregulated GO term.

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{2}
\tightlist
\item
  Which of the shown GO terms has the highest GeneRatio ?
\end{enumerate}

\begin{center}\includegraphics{Task-3_files/figure-latex/unnamed-chunk-26-1} \end{center}

\textbf{Answer:} By looking at the graphs,which are sorted by GeneRatio,
we can see that \textbf{a gene cluster with mobility functions like cell
motility or flagellum-dependent cell motility} rank highest in terms of
GeneRatio.

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\setcounter{enumi}{3}
\tightlist
\item
  Which of the shown GO terms has the highest Count ?
\end{enumerate}

\begin{center}\includegraphics{Task-3_files/figure-latex/unnamed-chunk-27-1} \end{center}

\textbf{Answer:} By looking at the graphs,which are sorted by Counts, we
can see that \textbf{the same gene cluster with mobility functions like
cell motility or flagellum-dependent cell motility} rank also highest in
terms of Counts.

\subsection{Clustering by GO
relations}\label{clustering-by-go-relations}

\begin{enumerate}
\def\labelenumi{\alph{enumi})}
\tightlist
\item
  How many clusters can you identify when showing 10, 20, and 30
  categories
\end{enumerate}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{enrichedBP }\OtherTok{\textless{}{-}}\NormalTok{ enrichplot}\SpecialCharTok{::}\FunctionTok{pairwise\_termsim}\NormalTok{(orUpBP)}
\FunctionTok{emapplot}\NormalTok{(enrichedBP, }\AttributeTok{showCategory =} \DecValTok{10}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{center}\includegraphics{Task-3_files/figure-latex/unnamed-chunk-28-1} \end{center}

\newpage

\section*{References}\label{references}
\addcontentsline{toc}{section}{References}

\phantomsection\label{refs}
\begin{CSLReferences}{0}{0}
\bibitem[\citeproctext]{ref-pollo2022absence}
\CSLLeftMargin{{[}1{]} }%
\CSLRightInline{L. Pollo-Oliveira \emph{et al.}, {``The absence of the
queuosine tRNA modification leads to pleiotropic phenotypes revealing
perturbations of metal and oxidative stress homeostasis in escherichia
coli K12,''} \emph{Metallomics}, vol. 14, no. 9, p. mfac065, 2022.}

\bibitem[\citeproctext]{ref-R-DESeq2}
\CSLLeftMargin{{[}2{]} }%
\CSLRightInline{M. Love, S. Anders, and W. Huber, \emph{DESeq2:
Differential gene expression analysis based on the negative binomial
distribution}. 2025. doi:
\href{https://doi.org/10.18129/B9.bioc.DESeq2}{10.18129/B9.bioc.DESeq2}.}

\bibitem[\citeproctext]{ref-R-ggplot2}
\CSLLeftMargin{{[}3{]} }%
\CSLRightInline{H. Wickham \emph{et al.}, \emph{ggplot2: Create elegant
data visualisations using the grammar of graphics}. 2025. Available:
\url{https://ggplot2.tidyverse.org}}

\bibitem[\citeproctext]{ref-R-knitr}
\CSLLeftMargin{{[}4{]} }%
\CSLRightInline{Y. Xie, \emph{Knitr: A general-purpose package for
dynamic report generation in r}. 2025. Available:
\url{https://yihui.org/knitr/}}

\bibitem[\citeproctext]{ref-R-rmarkdown}
\CSLLeftMargin{{[}5{]} }%
\CSLRightInline{J. Allaire \emph{et al.}, \emph{Rmarkdown: Dynamic
documents for r}. 2025. Available:
\url{https://github.com/rstudio/rmarkdown}}

\bibitem[\citeproctext]{ref-R-tidyverse}
\CSLLeftMargin{{[}6{]} }%
\CSLRightInline{H. Wickham, \emph{Tidyverse: Easily install and load the
tidyverse}. 2023. Available: \url{https://tidyverse.tidyverse.org}}

\bibitem[\citeproctext]{ref-DESeq22014}
\CSLLeftMargin{{[}7{]} }%
\CSLRightInline{M. I. Love, W. Huber, and S. Anders, {``Moderated
estimation of fold change and dispersion for RNA-seq data with
DESeq2,''} \emph{Genome Biology}, vol. 15, p. 550, 2014, doi:
\href{https://doi.org/10.1186/s13059-014-0550-8}{10.1186/s13059-014-0550-8}.}

\bibitem[\citeproctext]{ref-ggplot22016}
\CSLLeftMargin{{[}8{]} }%
\CSLRightInline{H. Wickham, \emph{ggplot2: Elegant graphics for data
analysis}. Springer-Verlag New York, 2016. Available:
\url{https://ggplot2.tidyverse.org}}

\bibitem[\citeproctext]{ref-knitr2015}
\CSLLeftMargin{{[}9{]} }%
\CSLRightInline{Y. Xie, \emph{Dynamic documents with {R} and knitr}, 2nd
ed. Boca Raton, Florida: Chapman; Hall/CRC, 2015. Available:
\url{https://yihui.org/knitr/}}

\bibitem[\citeproctext]{ref-knitr2014}
\CSLLeftMargin{{[}10{]} }%
\CSLRightInline{Y. Xie, {``Knitr: A comprehensive tool for reproducible
research in {R},''} in \emph{Implementing reproducible computational
research}, V. Stodden, F. Leisch, and R. D. Peng, Eds., Chapman;
Hall/CRC, 2014.}

\bibitem[\citeproctext]{ref-rmarkdown2018}
\CSLLeftMargin{{[}11{]} }%
\CSLRightInline{Y. Xie, J. J. Allaire, and G. Grolemund, \emph{R
markdown: The definitive guide}. Boca Raton, Florida: Chapman; Hall/CRC,
2018. Available: \url{https://bookdown.org/yihui/rmarkdown}}

\bibitem[\citeproctext]{ref-rmarkdown2020}
\CSLLeftMargin{{[}12{]} }%
\CSLRightInline{Y. Xie, C. Dervieux, and E. Riederer, \emph{R markdown
cookbook}. Boca Raton, Florida: Chapman; Hall/CRC, 2020. Available:
\url{https://bookdown.org/yihui/rmarkdown-cookbook}}

\bibitem[\citeproctext]{ref-tidyverse2019}
\CSLLeftMargin{{[}13{]} }%
\CSLRightInline{H. Wickham \emph{et al.}, {``Welcome to the
{tidyverse},''} \emph{Journal of Open Source Software}, vol. 4, no. 43,
p. 1686, 2019, doi:
\href{https://doi.org/10.21105/joss.01686}{10.21105/joss.01686}.}

\end{CSLReferences}

\end{document}
